### PR TITLE
refactor: use DnsRecord in code and  DNS Record in UI text

### LINKED ISSUE #245: Standardize on naming for Domain vs. Record
All over the code we use Record and Domain in various contexts.  I'm reviewing code right now that does this:

```ts
function onDomainAction(domain: Record, action: DomainsTableAction) {
```

This seems odd to me.  Can we come-up with a standard way to deal with this through the code and normalize our types/names?

### PR DESCRIPTION
**Pull request by SerpentBytes**
## Description
Resolves #245 by replacing instances of [Dd]omain(s) or [Rr]ecord(s) referencing to DNS Records to [dD]nsRecord(s) in code and to DNS Record(s) in UI text. 

### Steps to Test
- Run the project locally
- Test out all DNS record routes
- Run tests (All Unit Tests / E2E tests must pass in the development/staging environment)

### CODE REVIEW

File: app/components/header.tsx [hunk 1]
```diff
@@ -37,10 +37,10 @@ export default function Header() {
       <Hide below="lg">
         <HStack color="white" justifyContent="flex-start" gap="10" width="100%">
           <Flex gap="5">
-            <Link to={{ pathname: '/domains' }}>
+            <Link to={{ pathname: '/dns-records' }}>
               <Flex alignItems="center">
                 <TriangleUpIcon marginRight="2" />
-                <Text>Domains</Text>
+                <Text>Records</Text>
```
Comment (Eakam1007) [inline app/components/header.tsx:43]: I think even in the UI, we should call these DNS Records

**Comment by SerpentBytes** [comment]
@Eakam1007 It seems like Playwright is running cached tests which contained the typo.

**Comment by Eakam1007** [comment]
> @Eakam1007 It seems like Playwright is running cached tests which contained the typo.

Hmmm, that's weird. As far as I know, we do not cache tests, only the installed browsers

**Comment by Eakam1007** [comment]
Could you please confirm by downloading the report ([specifics here](https://github.com/DevelopingSpace/starchart/wiki/Playwright#debugging-ci-failures)) ?

**Comment by SerpentBytes** [comment]
> Could you please confirm by downloading the report ([specifics here](https://github.com/DevelopingSpace/starchart/wiki/Playwright#debugging-ci-failures)) ?

I was unable to find the report. Let me check.

**Comment by Eakam1007** [comment]
You can find the report at the bottom of the summary page of the CI run, here: https://github.com/DevelopingSpace/starchart/actions/runs/4473175500?pr=397

**Comment by Eakam1007** [comment]
@SerpentBytes could you please update the PR title and description to reflect that we use `DNS Records` in the UI as well?

File: app/lib/dns.server.ts [hunk 2]
```diff
@@ -114,40 +114,45 @@ export async function createHostedZone(domain: string) {
   }
 }
 
-export const createRecord = (username: string, type: RecordType, name: string, value: string) => {
+export const createDnsRecord = (
+  username: string,
+  type: DnsRecordType,
+  name: string,
```
Comment (humphd) [inline app/lib/dns.server.ts:120]: Noticing we still use `name` here vs. `subdomain`, filed https://github.com/DevelopingSpace/starchart/issues/399 to deal with that later.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -15,25 +15,25 @@ model User {
...
   updatedAt   DateTime      @updatedAt
   deactivated Boolean       @default(false)
-  record      Record[]
+  dnsRecord   DnsRecord[]
   certificate Certificate[]
 }
 
-model Record {
-  id           Int          @id @default(autoincrement())
+model DnsRecord {
```
Comment (humphd) [inline prisma/schema.prisma:22]: When this is merged, ping me to figure out how we'll migrate the staging db.

cc @dadolhay.

**Review by humphd — APPROVED** [review]
Epic

File: app/lib/dns.server.ts (hunk 2 above)
Comment (SerpentBytes) [reply app/lib/dns.server.ts:120]: My bad. I missed a file.

Comment (humphd) [reply app/lib/dns.server.ts:120]: No, this isn't anything related to you.  This is from work we were doing last week to rename things in the DNS code.

File: app/models/dns-record.server.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  });
+
+  return count > 0;
+}
+
+export function deleteDnsRecordById(id: DnsRecord['id']) {
+  return prisma.dnsRecord.delete({ where: { id } });
+}
+/* duplicate function */
+// export function renewDnsRecordById(id: DnsRecord['id']) {
```
Comment (Genne23v) [inline app/models/dns-record.server.ts:112]: If you are looking for a confirmation, you can remove this duplicate function.

File: app/queues/dns/add-record-flow.server.ts [hunk 5]
```diff
@@ -4,17 +4,17 @@ import { buildDomain } from '~/utils';
 import { dnsUpdateQueueName } from './workers/dns-update-worker.server';
```
Comment (Genne23v) [inline app/queues/dns/add-record-flow.server.ts:4]: Maybe we should also change this file name

**Review by Genne23v — CHANGES_REQUESTED** [review]

File: app/lib/dns.server.ts (hunk 2 above)
Comment (Genne23v) [reply app/lib/dns.server.ts:120]: I will quickly update as soon as this PR gets merged.

**Review by Genne23v — APPROVED** [review]
LGTM!

**Review by sfrunza13 — APPROVED** [review]

**Review by Eakam1007 — APPROVED** [review]

**Review by Ririio — APPROVED** [review]
No issue with me

**Review by dadolhay — APPROVED** [review]
Nice, let's merge this as soon as possible so there is no conflict with other prs

**Comment by SerpentBytes** [comment]
Thanks for the reviews!

**Comment by humphd** [comment]
@dadolhay @cychu42 I need a plan for migrating the db on staging...

**Comment by humphd** [comment]
> Thanks for the reviews!

Thanks for doing it, we really needed this to happen!

**Review by cychu42 — COMMENTED** [review]
I know this is merged already, but this LGTM.

**Pull request merged**